### PR TITLE
Dependencies and Providers

### PR DESCRIPTION
**Pull request by dex-sv**

### CODE REVIEW

File: meta_aggregation_api/services/limit_orders.py [hunk 1]
```diff
@@ -60,28 +70,28 @@ async def get_by_wallet_address(
...
         if not provider_class:
             raise ProviderNotFound(provider)
-        provider_instance = provider_class()
+        provider_instance = provider_class(self.session, self.config, self.apm_client)
         logger.info(
             f'Getting limit order by hash: {order_hash}',
             extra={'provider': provider.__class__.__name__, 'order_hash': order_hash},
         )
-        res = await provider_instance.get_order_by_hash(
+        res = await provider_instance.get_order_by_hash(  # TODO: add the method to the base class?
```
Comment (dex-sv) [inline meta_aggregation_api/services/limit_orders.py:87]: @shestakoven 

This method is not a member of the BaseProvider ABC. We should either add it to the BaseProvider or perform a type check on the provider instance.

File: meta_aggregation_api/services/limit_orders.py [hunk 2]
```diff
@@ -91,12 +101,12 @@ async def post(
...
         if not provider_class:
             raise ProviderNotFound(provider)
-        provider_instance = provider_class()
+        provider_instance = provider_class(self.session, self.config, self.apm_client)
         logger.info(
             f'Posting limit order: {order_hash}',
             extra={'provider': provider.__class__.__name__, 'order_hash': order_hash},
         )
-        response = await provider_instance.post_limit_order(
+        response = await provider_instance.post_limit_order(  # TODO: add the method to the base class?
```
Comment (dex-sv) [inline meta_aggregation_api/services/limit_orders.py:109]: @shestakoven Unresolved attribute reference 'get_order_by_hash' for class 'BaseProvider'

**Review by safor — APPROVED** [review]
LGTM

File: meta_aggregation_api/services/limit_orders.py (hunk 1 above)
Comment (evahteev) [reply meta_aggregation_api/services/limit_orders.py:87]: that's the method used only for limit orders type of provider... it's not needed for the rest of the providers. let's add to Base

File: meta_aggregation_api/services/limit_orders.py (hunk 2 above)
Comment (evahteev) [reply meta_aggregation_api/services/limit_orders.py:109]: ^

File: meta_aggregation_api/services/limit_orders.py (hunk 1 above)
Comment (dex-sv) [reply meta_aggregation_api/services/limit_orders.py:87]: Why not just remove the `provider` parameter from the method and explicitly use LimitOrgerService class ?

File: meta_aggregation_api/rest_api/create_app.py [hunk 3]
```diff
@@ -50,21 +54,59 @@ def create_app(config: Config):
...
     )
     providers = dependencies.ProvidersConfig()
-    deps = dependencies.Dependencies(
-        aiohttp_session=aiohttp_session,
-        config=config,
-        chains=chains,
-        gas_service=gas_service,
-        limit_orders_service=dependencies.LimitOrdersService(config=config),
-        meta_aggregation_service=dependencies.MetaAggregationService(
+    prorvider_registry = dependencies.ProviderRegistry(
```
Comment (dex-sv) [inline meta_aggregation_api/rest_api/create_app.py:57]: Providers as dependencies

File: meta_aggregation_api/config/__init__.py [hunk 4]
```diff
@@ -15,7 +15,7 @@ class Config(APMConfig, LoggerConfig, AuthConfig, CacheConfig, BaseSettings):
     NATIVE_TOKEN_ADDRESS = '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee'
     VERSION = '0.0.1'
     PUBLIC_KEY = 'Get your key at https://developers.dex.guru/'
-    PUBLIC_API_DOMAIN = 'https://public-stage-lax.dexguru.biz/'
+    PUBLIC_API_DOMAIN: HttpUrl = 'https://public-stage-lax.dexguru.biz/'
```
Comment (shestakoven) [inline meta_aggregation_api/config/__init__.py:18]: Remove stage from here

File: meta_aggregation_api/providers/zerox_v1/zerox_provider.py [hunk 5]
```diff
@@ -240,10 +240,11 @@ async def get_swap_quote(
 
     async def get_orders_by_trader(
         self,
-        taker_token: str,
-        maker_token: str,
+        *,
+        chain_id: str,
```
Comment (shestakoven) [inline meta_aggregation_api/providers/zerox_v1/zerox_provider.py:244]: Chain id type str?

File: meta_aggregation_api/config/__init__.py (hunk 4 above)
Comment (dex-sv) [reply meta_aggregation_api/config/__init__.py:18]: What to put instead?

File: meta_aggregation_api/providers/zerox_v1/zerox_provider.py (hunk 5 above)
Comment (dex-sv) [reply meta_aggregation_api/providers/zerox_v1/zerox_provider.py:244]: this got inferred by PyCharm from the `_api_path_builder(..., chain_id: str)` below

Comment (dex-sv) [reply meta_aggregation_api/providers/zerox_v1/zerox_provider.py]: fixed

File: meta_aggregation_api/services/limit_orders.py (hunk 2 above)
Comment (dex-sv) [reply meta_aggregation_api/services/limit_orders.py]: fixed in the latest commits

**Review by evahteev — APPROVED** [review]

**Pull request merged**